llvm/lib/CodeGen/RegisterCoalescer.cpp

```diff
@@ -1941,6 +1942,7 @@ void RegisterCoalescer::updateRegDefsUses(Register SrcReg, Register DstReg,
       if (SubIdx && MO.isDef()) {
         MO.setIsUndef(!Reads);
         FullDef = false;
+        DeadDef = MO.isDead();
```

Comment:
I don't think dead flags are required to be accurate, it might be safer to check if LiveIntervals thinks it's really daed